enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,364 @@
+- It does not address certificate management for endpoints other than the API endpoint (`api.cluster.example.com`).
+- It does not provide a solution for environments where `nftables` is not supported.
+
+## Proposal
+
+The HTTP01 Challenge Proxy will be implemented as a **core component of the OpenShift payload** and managed by an existing cluster operator, specifically the **cluster-kube-apiserver-operator**, which already manages API endpoint configuration and lifecycle.
+
+**Deployment Architecture:**
+- **Core Integration**: This feature is part of the core OpenShift distribution, not an optional OLM-installable operator
+- **Operator Management**: The cluster-kube-apiserver-operator will deploy and manage the proxy DaemonSet
+- **Conditional Deployment**: The proxy is only deployed when `APIServer.spec.http01ChallengeProxy.enabled = true`
+- **Platform Targeting**: Limited to baremetal platforms where HTTP01 challenges are needed for API certificates
+
+**Operational Responsibilities:**
+- Deploy and manage a DaemonSet running on control plane nodes that may host the API VIP
+- **Traffic Filtering**: Only redirect HTTP traffic destined for `<API_VIP>/.well-known/acme-challenge/*` paths (ACME HTTP01 challenge traffic)
+- **Path Validation**: Use path-based filtering rather than SNI (Server Name Indication) to identify HTTP01 challenge requests
+- Redirect filtered HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers  
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`
+- Handle lifecycle management, upgrades, and configuration of the proxy
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+Rather than creating a new CRD, the HTTP01 Challenge Proxy configuration will be added as a new field to the existing **APIServer** CRD in the [openshift/api](https://github.com/openshift/api) repo. This approach aligns with the pattern used by other API server features like audit, encryption, and TLS configuration.
+
+The configuration will be added to the `APIServerSpec` struct and managed by the cluster-kube-apiserver-operator as part of its normal reconciliation loop. Since the APIServer CRD is already a singleton (named "cluster"), no additional enforcement mechanisms are needed.
+
+**API Changes to config/v1/types_apiserver.go:**
+
+```go
+type APIServerSpec struct {
+    // ... existing fields ...
+    
+    // http01ChallengeProxy contains configuration for the HTTP01 challenge proxy
+    // that redirects traffic from the API endpoint on port 80 to ingress routers.
+    // This enables cert-manager to perform HTTP01 ACME challenges for API endpoint certificates.
+    // +optional
+    HTTP01ChallengeProxy *HTTP01ChallengeProxySpec `json:"http01ChallengeProxy,omitempty"`
+}
+
+type HTTP01ChallengeProxySpec struct {
```

Comment:
Because the parent field is already optional, you'll want at least one of these children fields to be required.